### PR TITLE
Allow for the submitting of flags after game end

### PR DESCRIPTION
**Pull request by zwade**
Adds a config option to enable submitting of flags after competition end, with a 0 point award for doing so.

Does not update UI, still shows the same "Game Over" message regardless.

### CODE REVIEW

**Comment by zwade** [comment]
Addresses #3

File: scoreboard/controllers.py [hunk 1]
```diff
@@ -98,12 +98,21 @@ def submit_answer(cid, answer):
...
+            else:
+                correct = 'CORRECT (Game Over)'
+
             team.last_solve = datetime.datetime.utcnow()
             models.ScoreHistory.add_entry(team)
             challenge.update_answers(exclude_team=team)
-            correct = 'CORRECT'
-            return ans.current_points
+
+            if not utils.GameTime.over():
```
Comment (Matir) [inline scoreboard/controllers.py]: Please swap this logic and remove the "not"

File: scoreboard/controllers.py [hunk 2]
```diff
@@ -98,12 +98,21 @@ def submit_answer(cid, answer):
             raise errors.AccessDeniedError('Challenge is locked!')
         if challenge.verify_answer(answer):
             ans = models.Answer.create(challenge, team, answer)
-            team.score += ans.current_points
+
+            if not utils.GameTime.over():
```
Comment (Matir) [inline scoreboard/controllers.py]: Please swap the logic and remove the not

File: scoreboard/controllers.py (hunk 1 above)
Comment (Matir) [inline scoreboard/controllers.py]: Same as above.

File: config.py [hunk 3]
```diff
@@ -21,3 +21,6 @@
 TEAMS = True
 ATTACHMENT_BACKEND = 'file:///tmp/attachments'
 LOGIN_METHOD = 'local'
+SESSION_COOKIE_SECURE = False
+GAME_TIME = ("2012-01-19 17:21:00", "2016-06-01 17:00:00")
+SUBMIT_AFTER_END = True
```
Comment (Matir) [inline config.py]: Please don't commit changes to config.py.

File: scoreboard/utils.py [hunk 4]
```diff
@@ -157,6 +158,14 @@ def open(cls, after_end=False):
         return False
 
     @classmethod
+    def over(cls):
+        """Has the game ended?"""
+        state = cls.state()
+        if state == 'AFTER':
```
Comment (Matir) [inline scoreboard/utils.py]: you can just:
return state == 'AFTER'

File: scoreboard/controllers.py [hunk 5]
```diff
@@ -99,20 +99,20 @@ def submit_answer(cid, answer):
...
                 team.score += ans.current_points
                 correct = 'CORRECT'
-            else:
-                correct = 'CORRECT (Game Over)'
 
             team.last_solve = datetime.datetime.utcnow()
             models.ScoreHistory.add_entry(team)
             challenge.update_answers(exclude_team=team)
 
             if not utils.GameTime.over():
```
Comment (Matir) [inline scoreboard/controllers.py]: You did not remove the not.

File: scoreboard/utils.py [hunk 6]
```diff
@@ -161,9 +161,7 @@ def open(cls, after_end=False):
     def over(cls):
         """Has the game ended?"""
         state = cls.state()
-        if state == 'AFTER':
-            return True
-        return False
+        return state == 'AFTER':
```
Comment (Matir) [inline scoreboard/utils.py]: No colon.

**Comment by zwade** [comment]
resolves #3

**Pull request merged**